lib/auth/presence/presencev1/service_test.go

```diff
@@ -184,14 +184,31 @@ func TestGetRemoteCluster(t *testing.T) {
 			client, err := srv.NewClient(auth.TestUser(tt.user))
 			require.NoError(t, err)
 
-			bot, err := client.PresenceServiceClient().GetRemoteCluster(ctx, tt.req)
+			rc, err := client.PresenceServiceClient().GetRemoteCluster(ctx, tt.req)
 			tt.assertError(t, err)
 			if tt.want != nil {
-				// Check that the returned bot matches
-				require.Empty(t, cmp.Diff(tt.want, bot, protocmp.Transform()))
+				// Check that the returned remote cluster matches
+				require.Empty(t, cmp.Diff(tt.want, rc, protocmp.Transform()))
 			}
 		})
 	}
+
+	t.Run("doesnt exist and no permissions errors match", func(t *testing.T) {
+		client, err := srv.NewClient(auth.TestUser(user.GetName()))
+		require.NoError(t, err)
+
+		_, doesntExistError := client.PresenceServiceClient().GetRemoteCluster(ctx, &presencev1pb.GetRemoteClusterRequest{
+			Name: "non-existent",
+		})
+		require.Error(t, doesntExistError)
+		_, noPermissionsError := client.PresenceServiceClient().GetRemoteCluster(ctx, &presencev1pb.GetRemoteClusterRequest{
+			Name: notMatchingRC.GetName(),
+		})
+		require.Error(t, noPermissionsError)
+
+		require.Equal(t, doesntExistError.Error(), noPermissionsError.Error(),
```

Comment:
Maybe add checks to confirm that they're also both a `NotFound`, or perhaps that they're equal error objects after a `trace.Unwrap`; I think it's possible to have errors with the same `.Error()` but that are transmitted with different codes (`trace.AccessDenied("not found")` and `trace.NotFound("not found")`).